entities/entities/drug_lab/init.lua

```diff
@@ -24,9 +25,9 @@ function ENT:canUse(activator)
 end
 
 function ENT:createItem(activator)
-    local drugPos = self:GetPos()
+    local drugPos = self:GetPos() + self.SpawnOffset
```

Comment:
Adding two vectors still creates a new vector, so it's not really an optimisation. I wouldn't make any further steps to optimise it though, the micro change would not be worth the decrease in readability.

This does have the nice effect of making the spawn offset configurable.

Reply:
Serverside functions which are called only once don't need this sort of optimisation.

Reply:
It is faster and also makes the code more readable in my opinion.

Ran 10000 times as smaller runs gives scientific notation
```
Vector(x,y,z + 30)
0.0072376109999368
Vec + Vec
0.0021952729999839
```

Reply:
That's adding a vector of itself. That might not give the same performance as adding two distinct vectors. Either way, these functions aren't being run 10,000 times, so there is no performance difference.

Like I said, though, damn the `0.0000005` seconds of performance difference for every drug lab creation. Being able to set the variable is nice enough.

Reply:
Vector:Add(Vector) preserves the original object and is readable. They are significantly faster operations than creating a new object which also adds all the vector pool operations.

Reply:
> That's adding a vector of itself. That might not give the same performance as adding two distinct vectors. Either way, these functions aren't being run 10,000 times, so there is no performance difference.
> 
> Like I said, though, damn the `0.0000005` seconds of performance difference for every drug lab creation. Being able to set the variable is nice enough.

Sorry I should have labeled them better. That is adding two distinct vectors.

```
local vec = Vector(100, 100, 100)
local vec2 = Vector(0,0,30)
local t

print("Vector(x,y,z + 30)")
local time = SysTime()
for i=1,10000 do
	t = Vector(vec.x,vec.y,vec.z + 30)
end
print(SysTime() - time)

print("Vec + Vec")
local time = SysTime()
for i=1,10000 do
	t = vec + vec2
end
print(SysTime() - time)
```

I did this mainly for the fact it had better readability over the previous one, with the added benefit you are able to change the offset. The performance benefits are just a nice side effect.